### PR TITLE
Colors Tests

### PR DESCRIPTION
**Pull request by computamike**
Sorry I took so long on this - I've not felt too great.

Adding tests for colors end point : 

- Fixed a bug : where colour route defaulted to max (100) id qty not specified.

unsure how to test error conditions using supertest, as this would require generating values outside the ranges specified in the colorspaces.ts file - which would be difficult to test using supertest.

### CODE REVIEW

**Comment by computamike** [comment]
@ageddesi - one functional thing we might need to consider : Passing in a nonsense quantity (in my tests - "foo") returns 100 records.  I think that when pulling the qty from the request we might be better to default to 1, or to throw an error.

Left a comment on the test - not sure how you would want to fix that - however the existing functionality is tested - so maybe this is an issue for someone to look at?

hope this is ok

File: modules/colors/tests/colors.test.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+describe('color api endpoints', () => {
+ 
+  describe('GET /color/', () => {
+    it('requesting with no qty returns a list of 1 color', async () => {
+      const qty = 1;
+      const response = await request(app).get(`/colors/`);
+      expect(response.body.length).toBe(qty);
+    });
+    it('requesting with non-numerical qty returns a list of 100 color', async () => {
+      // I think this is potentially a bug - we probably should be retuning 1 colour, or throwing an error.
```
Comment (ageddesi) [inline modules/colors/tests/colors.test.ts:12]: @computamike Can you raise this as a bug. Thanks. We should be doing a check in that util to make sure its a number and defaulting if it is not.

File: utils/route-utils.ts [hunk 2]
```diff
@@ -7,14 +7,33 @@ const imageWidthDefault = 640;
...
+
+    return  request.params.qty ? 
+                isNumber(request.params.qty) && parseInt(request.params.qty) < maxQty ? 
+                    parseInt(request.params.qty)
+                    : maxQty
+                :defaultValue
+
+ 
+
+    // return request.params.qty
```
Comment (ageddesi) [inline utils/route-utils.ts:30]: If you can remove the commented out code :)

**Review by ageddesi — CHANGES_REQUESTED** [review]

File: modules/colors/tests/colors.test.ts (hunk 1 above)
Comment (computamike) [reply modules/colors/tests/colors.test.ts:12]: have raised #239 for this

File: utils/route-utils.ts (hunk 2 above)
Comment (computamike) [reply utils/route-utils.ts]: that should be done now

**Review by ageddesi — APPROVED** [review]

**Comment by ageddesi** [comment]
Wow, what a ride, Thanks @computamike 👍

**Pull request merged**